### DISCUSSION TITLE
Wemos D1 Mini with D1 Relay Module - Firmware Update Issue

### DISCUSSION THREAD

**Discussion by cgrylmz** [RavenSystem/esp-homekit-devices#2331] [MEPLHAA Scripts]
Hey all, 
I've been using Wemos D1 Mini and D1 Relay Module. See below;
![D1 Relay Module](https://github.com/user-attachments/assets/092cbc84-19dd-456b-b372-43069fb14731)
![Wemos D1 Mini](https://github.com/user-attachments/assets/baaf0097-dedc-4bc6-a92a-da13d8e1b652)

I've been using this script to use it as switch which relay is closed after 1,5 seconds it opens everytime. (I'm using it as a door automat. 

My Script: 
{
  "c": {
    "l": 2
  },
  "a": [
    {
      "0": {
        "r": [
          {
            "g": 5,
            "v": 0
          }
        ]
      },
      "1": {
        "r": [
          {
            "g": 5,
            "v": 1
          }
        ]
      },
      "t": 1,
      "b": [
        {
          "g": 0,
          "t": 1
        }
      ],
      "s": 0,
      "i": 1.5
    }
  ]
}

I've recently updated the firmware by going in setup mode and checking the firmware update box. Cannot remember clearly but, checked box which was saying something like "Update firmware for all" something like that, i'm not sure. 

After update is complete, i could see the device in homekit, i've checked the firmware version, it is updated. Can get it to main firmware, it looks like its updated. I can turn on the device (and it turns off after 1,5 seconds as it should) on homekit but nothing is happening to relay, no sound, as it should have. 

I was thinking maybe with the update, number of gpios was changed? Is it possible? 

Couldn't solve the issue. Thinking to flash and re-install everything, but i don't know if it helps. Maybe i need to update meplhaa script too?

Any help would be appriciated.

**Comment by RavenSystem** [comment]
https://github.com/RavenSystem/esp-homekit-devices/wiki/migration-guide-for-haa-v12-merlin

```json
{"c":{"io":[[[5,2],2],[[0],6]],"l":2,"b":[[0,5]]},"a":[{"0":{"r":[[5]]},"1":{"r":[[5,1]]},"b":[[0]],"i":1.5}]}
```

**Comment by cgrylmz** [comment]
Thank you! Worked.
